mp/src/game/client/hud_crosshair.cpp

```diff
@@ -143,164 +186,393 @@ extern ConVar cl_crosshair_scale;
+
+    int vx, vy, vw, vh;
+    vgui::surface()->GetFullscreenViewport( vx, vy, vw, vh );
+
+    if (cl_crosshair_style.GetInt() == 0) //only CS:S uses crosshair scaling
+    {
+        int crosshairScale = cl_crosshair_scale.GetInt();
+
+        if (crosshairScale < 1)
+        {
+            if (vh <= 600)
+                crosshairScale = 600;
+            else if (vh <= 768)
+                crosshairScale = 768;
+            else
+                crosshairScale = 1200;
+        }
+
+        if (cl_crosshair_scale_enable.GetBool() == false)
+            scale = 1.0f;
+        else
+            scale = float(vh) / float(crosshairScale);
+
+        iCrosshairDistance = static_cast<int>(ceil(weaponBase->m_flCrosshairDistance * scale));
+        iBarSize = XRES(5) + (iCrosshairDistance - iDistance) / 2;
+        iBarSize = max(1, (int)((float)iBarSize * scale));
+        iBarThickness = max(1, (int)floor(scale + 0.5f)); //thickness of 1 (or odd) causes off-center crosshairs
+    }
+    else //allow user cvars
+    {
+        scale = 1.0f;
+
+        iCrosshairDistance = static_cast<int>(ceil(weaponBase->m_flCrosshairDistance * scale));
+        iBarSize = cl_crosshair_size.GetInt();
+        iBarThickness = cl_crosshair_thickness.GetInt(); //thickness of 1 (or odd) causes off-center crosshairs
+    }
+    
+    uint32 newHexColor = MomUtil::GetHexFromColor(cl_crosshair_color.GetString());
+    Color crossColor;
+    if (!MomUtil::GetColorFromHex(newHexColor, crossColor))
+        crossColor.SetColor(255, 0, 0, 200);
+    if (!cl_crosshair_alpha_enable.GetBool())
+        crossColor.SetColor(crossColor.r(), crossColor.g(), crossColor.b(), 200);
+
+    int iHalfScreenWidth = vw / 2;
+    int iHalfScreenHeight = vh / 2;
+
+    if (cl_crosshair_style.GetInt() != 2)
+    {
+        m_pCrosshair = gHUD.GetIcon("whiteAdditive");
+
+        if (m_pCrosshair)
+            weaponBase->m_iCrosshairTextureID = m_pCrosshair->textureId;
+        
+        int iLeft = iHalfScreenWidth - (iCrosshairDistance + iBarSize);
+        int iRight = iHalfScreenWidth + iCrosshairDistance;
+        int iFarLeft = iLeft + iBarSize;
+        int iFarRight = iRight + iBarSize;
+
+        int iTop = iHalfScreenHeight - (iCrosshairDistance + iBarSize);
+        int iBottom = iHalfScreenHeight + iCrosshairDistance;
+        int iFarTop = iTop + iBarSize;
+        int iFarBottom = iBottom + iBarSize;
+
+        int iHalfUpper = iHalfScreenHeight - iBarThickness / 2;
+        int iHalfLower = iHalfScreenHeight + iBarThickness / 2 + iBarThickness % 2;
+        int iHalfLefter = iHalfScreenWidth - iBarThickness / 2;
+        int iHalfRighter = iHalfScreenWidth + iBarThickness / 2 + iBarThickness % 2;
+
+        int iOutlineThickness = cl_crosshair_outline_thickness.GetInt();
+
+        if (!cl_crosshair_alpha_enable.GetBool())
+        {
+            // Additive crosshair
+            vgui::surface()->DrawSetTexture(weaponBase->m_iCrosshairTextureID);
```

Comment:
Since there's a `using namespace vgui;` up top you can remove the `vgui::` in this file.